configure.ac

```diff
@@ -128,19 +128,44 @@ AC_DEFUN([OMPI_CHECK_DIR_FOR_SPACES],[
            AC_MSG_ERROR([Cannot continue.])])
 ])
 
+dnl $1 = path
+dnl $2 = variable to which to assign canonicalized result
 AC_DEFUN([OMPI_CANONICALIZE_PATH],[
+    OPAL_VAR_SCOPE_PUSH([OMPI_ST OMPI_CP_PYTHON])
     case $host_os in
     darwin*)
-        # MacOS does not have "readlink -f" or realpath (at least as
-        # of MacOS Cataline / 10.15).  Instead, use Python, because we
-        # know MacOS comes with a /usr/bin/python that has
-        # os.path.realpath.
-        $2=`/usr/bin/python -c 'import os; print os.path.realpath("'$1'")'`
+        # Data points:
+        #
+        # - MacOS Catalina / 10.15 does not support the "-f" option to
+        #   readlink (to canonicalize filenames), and does not include
+        #   "realpath" in the base OS.
+        # - MacOS Monterrey / 12.3 does support "readlink -f", but
+        #   "realpath" is still not in the base OS.
+        #
+        # Instead of trying to figure out if "readlink -f" is
+        # supported, use Python: MacOS will have either
+        # /usr/bin/python (e.g., MacOS Catalina / 10.15) or
+        # /usr/bin/python3 (e.g., MacOS Monterrey / 12.3), both of
+        # which have os.path.realpath.
+        AS_IF([test -x /usr/bin/python],
```

Comment:
Ralph and I worked through this in OpenPMIx and came up with something way simpler: https://github.com/openpmix/openpmix/blob/7bd5c452b302f656a8dd5c8e2a98b92a545143b0/configure.ac#L136

Reply:
There is no `python` executable in MacOS 12.3.1 Monterrey:

```
$ git clone git@github.com:openpmix/openpmix.git
...snipped...
$ cd openpmix
$ ./autogen.pl
...snipped...
$ ./configure
: builddir: /Users/jsquyres/git/openpmix
: srcdir: /Users/jsquyres/git/openpmix

============================================================================
== Configuring PMIx
============================================================================
checking build system type... x86_64-apple-darwin21.4.0
...snipped...
checking directory of source tree... .
./configure: line 6702: python: command not found
checking directory of absolute source tree... 
./configure: line 6734: python: command not found
checking directory of build tree... 
checking directory of prefix... /Users/jsquyres/bogus
./configure: line 6784: python: command not found
checking directory of absolute prefix... 
```

Reply:
@bwbarrett and I talked out of band: this commit now removes all the path-canonicalizing stuff (technically, it wasn't needed in the first place).  That avoids the issue altogether.